dlrover/python/master/hyperparams/simple_strategy_generator.py

```diff
@@ -81,68 +62,78 @@ def generate_opt_strategy(
                     gpu_stats, model_config, paral_config.dataloader
                 )
                 optimizer_config = self._generate_optimizer_config()
-                paral_configs[node.name] = ParallelConfig(
+                paral_configs[node.id] = ParallelConfig(
                     data_loader_config, optimizer_config
                 )
-        return paral_configs["simple_node"]
+                node.paral_config = paral_configs[node.id]
+        if paral_configs == {}:
+            logger.info("paral_configs length is 0")
+            return None
+        else:
+            logger.info(f"paral_configs: {paral_configs}")
+            return paral_configs[0]
 
     def _generate_dataloader_config(
         self, gpu_stats, model_config, dataloader_config
     ):
         if gpu_stats == []:
-            return DataLoaderConfig(0, "", 0, 0, 0)
+            return dataloader_config
         # Calculate the minimum remaining memory among GPUs
         min_remain_memory = min(
-            entry["total_memory_gb"] - entry["used_memory_gb"]
-            for entry in gpu_stats
+            entry.total_memory_mb - entry.used_memory_mb for entry in gpu_stats
         )
-
-        # Update dataloader configuration version
-        updated_version = dataloader_config.version + 1
-        # Extract dataloader configuration values
-        batch_size = dataloader_config.batch_size
-
-        # Extract model configuration values
-        block_size = model_config["block_size"]
-        n_layer = model_config["n_layer"]
-        n_heads = model_config["n_heads"]
-        n_embd = model_config["n_embd"]
-
-        # Calculate the memory required for intermediate activation
-        activation_memory_gb = (
-            (
-                34 * batch_size * block_size * n_embd
-                + 5 * batch_size * (block_size**2) * n_heads
+        if min_remain_memory > 2400:
```

Comment:
What's the unit of 2400, MB? It is better to add an annotation.
Why is the value 2400?

Reply:
To avoid the case that the remaining memory is too small and crashes into OOM. We set the minimum remaining memory to greater than 2400 MB. (1/10 of RTX 4090 vmem).

Reply:
TODO: We need to replace 2400 with a more reasonable value.

Reply:
A new issue #704 was created for this problem.